daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -146,39 +143,37 @@ private[lf] final class Compiler(
   }
 
   @inline
-  private[this] def nextPositionWithExprName(name: ExprVarName): Position = {
-    val p = nextPosition()
-    addExprVar(name, p)
-    p
-  }
-
-  private[this] def nextPositionWithOptionalExprName(name: Option[ExprVarName]): Position =
-    name.fold(nextPosition())(nextPositionWithExprName)
-
   private[this] def svar(p: Position): SEVar = SEVar(env.position - p.idx)
 
+  @inline
   private[this] def addVar(ref: VarRef, position: Position) =
     env = env.copy(varIndices = env.varIndices.updated(ref, position))
 
+  @inline
   private[this] def addExprVar(name: ExprVarName, position: Position) =
     addVar(EVarRef(name), position)
 
+  @inline
   private[this] def addTypeVar(name: TypeVarName, position: Position) =
     addVar(TVarRef(name), position)
 
+  @inline
   private[this] def hideTypeVar(name: TypeVarName) =
     env = env.copy(varIndices = env.varIndices - TVarRef(name))
 
   private[this] def vars: List[VarRef] = env.varIndices.keys.toList
 
+  @inline
   private[this] def lookupVar(varRef: VarRef): Option[SEVar] =
     env.varIndices.get(varRef).map(svar)
 
-  def lookupExprVar(name: ExprVarName): SEVar =
+  @inline
+  private[this] def lookupExprVar(name: ExprVarName): SEVar =
     lookupVar(EVarRef(name))
       .getOrElse(throw CompilationError(s"Unknown variable: $name. Known: ${vars.mkString(",")}"))
 
-  def lookupTypeVar(name: TypeVarName): Option[SEVar] =
+  @inline
```

Comment:
I'm always dubious about the value of adding `@inline` especially as we have no perf stats to indicate they actually have any benefit, or even that they don't harm. We just seem to add them sometimes when we think the methods look _small enough_. But surely the JVM compiler would be the better judge of that :shrug:

Reply:
I remove the one I added.